homeassistant/components/ebusd/__init__.py

```diff
@@ -0,0 +1,132 @@
+"""
+Support for Ebusd daemon for communication with eBUS heating systems.
+
+For more details about ebusd deamon, please refer to the documentation at
+https://github.com/john30/ebusd
+"""
+
+from datetime import timedelta
+import logging
+import socket
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT, CONF_MONITORED_CONDITIONS)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.util import Throttle
+
+from .const import (
+    DOMAIN, SENSOR_TYPES, READ_COMMAND, WRITE_COMMAND)
+
+REQUIREMENTS = ['ebusdpy==0.0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_EBUSD = 'EBUSD'
+DEFAULT_NAME = 'ebusd'
+DEFAULT_PORT = 8888
+CONF_CIRCUIT = 'circuit'
+CACHE_TTL = 900
+SERVICE_EBUSD_WRITE = 'ebusd_write'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=15)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CIRCUIT): cv.string,
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): vol.All(
+            cv.ensure_list, [vol.In(SENSOR_TYPES['700'])])
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the ebusd component."""
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+    name = conf.get(CONF_NAME)
+    circuit = conf.get(CONF_CIRCUIT)
+    monitored_conditions = conf.get(CONF_MONITORED_CONDITIONS)
+    server_address = (
+        conf.get(CONF_HOST), conf.get(CONF_PORT))
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        hass.data[DATA_EBUSD] = EbusdData(server_address, circuit)
+
+        sock.settimeout(5)
+        sock.connect(server_address)
+        sock.close()
+
+        sensor_config = {
+            'monitored_conditions': monitored_conditions,
+            'client_name': name,
+            'sensor_types': SENSOR_TYPES[circuit]
+        }
+        load_platform(hass, 'sensor', DOMAIN, sensor_config, config)
+
+        hass.services.register(
+            DOMAIN, SERVICE_EBUSD_WRITE, hass.data[DATA_EBUSD].write)
+
+        _LOGGER.debug("Ebusd component setup completed.")
+        return True
+    except socket.timeout:
```

Comment:
The library should raise its own exceptions and make those available for import to the user of the library.